server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -183,31 +169,16 @@ public static ComposableIndexTemplate lookupTemplateForDataStream(String dataStr
     }
 
     public static void validateTimestampFieldMapping(String timestampFieldName, MapperService mapperService) {
-        MappedFieldType timestampFieldMapper = mapperService.fieldType(timestampFieldName);
-        if (timestampFieldMapper == null) {
-            throw new IllegalArgumentException("expected timestamp field [" + timestampFieldName + "], but found no timestamp field");
-        }
-        String type = timestampFieldMapper.typeName();
-        if (ALLOWED_TIMESTAMPFIELD_TYPES.contains(type) == false) {
-            throw new IllegalArgumentException("expected timestamp field [" + timestampFieldName + "] to be of types " +
-                ALLOWED_TIMESTAMPFIELD_TYPES + ", but instead found type [" + type  + "]");
-        }
-    }
+        TimestampFieldMapper fieldMapper = (TimestampFieldMapper) mapperService.documentMapper().mappers().getMapper("_timestamp");
+        assert fieldMapper != null : "[_timestamp] meta field mapper must exist";
 
-    public static String convertFieldPathToMappingPath(String fieldPath) {
-        // The mapping won't allow such fields, so this is a sanity check:
-        assert Arrays.stream(fieldPath.split("\\.")).filter(String::isEmpty).count() == 0L ||
-            fieldPath.startsWith(".") ||
-            fieldPath.endsWith(".") : "illegal field path [" + fieldPath + "]";
-
-        String mappingPath;
-        if (fieldPath.indexOf('.') == -1) {
-            mappingPath = "properties." + fieldPath;
-        } else {
-            mappingPath = "properties." + fieldPath.replace(".", ".properties.");
+        if (timestampFieldName.equals(fieldMapper.getPath()) == false) {
+            throw new IllegalArgumentException("[_timestamp] meta field doesn't point to data stream timestamp field [" +
```

Comment:
I think this error will be hard for end users to understand, since they shouldn't be aware of the `_timestamp` field. Maybe we won't even need this check, once we remove the ability to set `timestamp_field`, as discussed in https://github.com/elastic/elasticsearch/pull/59317#issuecomment-656236960?

Reply:
This is more of a sanity check. Kind of ensuring that we did apply the `_data_stream_timestamp` meta field mapper. If a user would run into this error then this would be a bug. Can tweak this check.